### PR TITLE
[AutoDiff] NFC: organize test/AutoDiff/activity_analysis.swift.

### PR DESCRIPTION
**Pull request by dan-zheng**
Categorize array literal and `inout` argument differentiation tests.
Add missing `CHECK` labels for `testArrayUninitializedIntrinsicFunctionResult`.
Add negative tests for TF-975 and TF-978.

---

TF-975 and TF-978 under investigation, fix coming soon.

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: test/AutoDiff/activity_analysis.swift [hunk 1]
```diff
@@ -121,91 +187,101 @@ func testArrayUninitializedIntrinsicAddress(_ x: Float, _ y: Float) -> [Float] {
...
+func testArrayUninitializedIntrinsicNested(_ x: Float, _ y: Float) -> [Float] {
+  let array = [x, y]
+  return [array[0], array[1]]
 }
+// CHECK-LABEL: [AD] Activity info for ${{.*}}testArrayUninitializedIntrinsicNested{{.*}} at (source=0 parameters=(0 1))
+// CHECK: [VARIED] %0 = argument of bb0 : $Float
+// CHECK: [VARIED] %1 = argument of bb0 : $Float
+// CHECK: [NONE]   %4 = integer_literal $Builtin.Word, 2
+// CHECK: [NONE]   // function_ref _allocateUninitializedArray<A>(_:)
+// CHECK: [VARIED]   %6 = apply %5<Float>(%4) : $@convention(thin) <τ_0_0> (Builtin.Word) -> (@owned Array<τ_0_0>, Builtin.RawPointer) // user: %7
```
Comment (dan-zheng) [inline test/AutoDiff/activity_analysis.swift]: Oops, `user:` comments should be removed.

```suggestion
// CHECK: [VARIED]   %6 = apply %5<Float>(%4) : $@convention(thin) <τ_0_0> (Builtin.Word) -> (@owned Array<τ_0_0>, Builtin.RawPointer)
```

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by marcrasi — APPROVED** [review]

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by rxwei — APPROVED** [review]

**Pull request merged**